servers/src/main/java/tachyon/worker/block/evictor/GreedyEvictor.java

```diff
@@ -0,0 +1,160 @@
+ * copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software distributed under the License
+ * is distributed on an "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express
+ * or implied. See the License for the specific language governing permissions and limitations under
+ * the License.
+ */
+
+package tachyon.worker.block.evictor;
+
+import java.io.IOException;
+import java.util.ArrayList;
+import java.util.HashMap;
+import java.util.List;
+import java.util.Map;
+
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+import com.google.common.base.Preconditions;
+
+import tachyon.Constants;
+import tachyon.Pair;
+import tachyon.worker.block.BlockMetadataManager;
+import tachyon.worker.block.BlockStoreEventListenerBase;
+import tachyon.worker.block.BlockStoreLocation;
+import tachyon.worker.block.meta.BlockMeta;
+import tachyon.worker.block.meta.StorageDir;
+import tachyon.worker.block.meta.StorageTier;
+
+/**
+ * A simple evictor that evicts arbitrary blocks until the required size is met. This class serves
+ * as an example to implement an Evictor.
+ */
+public class GreedyEvictor extends BlockStoreEventListenerBase implements Evictor {
+  private static final Logger LOG = LoggerFactory.getLogger(Constants.LOGGER_TYPE);
+  private final BlockMetadataManager mMetaManager;
+
+  public GreedyEvictor(BlockMetadataManager metadata) {
+    mMetaManager = Preconditions.checkNotNull(metadata);
+  }
+
+  @Override
+  public EvictionPlan freeSpace(long availableBytes, BlockStoreLocation location)
+      throws IOException {
+    // 1. Select a StorageDir that has enough capacity for required bytes.
+    StorageDir selectedDir = null;
+    if (location.equals(BlockStoreLocation.anyTier())) {
+      selectedDir = selectDirToEvictBlocksFromAnyTier(availableBytes);
+    } else {
+      int tierAlias = location.tierAlias();
+      StorageTier tier = mMetaManager.getTier(tierAlias);
+      if (location.equals(BlockStoreLocation.anyDirInTier(tierAlias))) {
+        selectedDir = selectDirToEvictBlocksFromTier(tier, availableBytes);
+      } else {
+        int dirIndex = location.dir();
+        StorageDir dir = tier.getDir(dirIndex);
+        if (canEvictBlocksFromDir(dir, availableBytes)) {
+          selectedDir = dir;
+        }
+      }
+    }
+    if (selectedDir == null) {
+      LOG.error("Failed to freeSpace: No StorageDir has enough capacity of {} bytes",
+          availableBytes);
+      return null;
+    }
+
+    // 2. Check if the selected StorageDir already has enough space.
+    List<Pair<Long, BlockStoreLocation>> toTransfer =
+        new ArrayList<Pair<Long, BlockStoreLocation>>();
+    List<Long> toEvict = new ArrayList<Long>();
+    long bytesAvailableInDir = selectedDir.getAvailableBytes();
+    if (bytesAvailableInDir >= availableBytes) {
```

Comment:
`canEvictBlocksFromDir()` is updated now and I don't think it can be used here any more.